need __version__.py if user needs to compile for their architecture or python version.

jamiecook opened the pull request.

jamiecook committed:
need __version__.py if user needs to compile for their architecture or python version.

pedrocamargo commented:
The paths for some of the pyx files changed (to ./cython), and that has to change in the setup.py as well. Something that @r-akemii can take a look at?

pedrocamargo commented on setup.py:
```diff
@@ -100,7 +100,7 @@
     "aequilibrae.reference_files": ["spatialite.sqlite", "nauru.zip", "sioux_falls.zip", "coquimbo.zip"],
     "aequilibrae.paths": ["parameters.pxi", "*.pyx"],
```
"aequilibrae.paths.cython": ["*.pix", "*.pyx" "*.pxd"],

r-akemii committed:
Update setup.py

r-akemii committed:
rename file path

pedrocamargo commented on aequilibrae/__init__.py:
```diff
@@ -11,7 +11,7 @@
 
 try:
-    from aequilibrae.paths.AoN import path_computation
+    from aequilibrae.paths.cython.AoN import path_computation
```
Wrong. The extension is NOT built inside the cython folder

pedrocamargo commented on aequilibrae/paths/__init__.py:
```diff
@@ -9,14 +9,14 @@
 from aequilibrae.paths.graph import Graph, TransitGraph
 from aequilibrae.paths.optimal_strategies import OptimalStrategies
-from aequilibrae.paths.public_transport import HyperpathGenerating
+from aequilibrae.paths.cython.public_transport import HyperpathGenerating
```
again

r-akemii committed:
removes cython from path

Commit:
.

pedrocamargo commented:
@Jake-Moss , can you cleanup this mess we made?  :-D

Jake-Moss commented:
Looks like I removed `__version__.py` from the package here https://github.com/AequilibraE/aequilibrae/commit/230d51ea04c4549a076e853a48b8136bc3bfc392 because of the request here https://github.com/conda-forge/staged-recipes/pull/22437#pullrequestreview-1980982844 . It was being packaged at the top level of wheel, which when unpacked would be outside of the `aequilibrae` package. The same was happening with our `test` dir but that should have been fixed via the `FlatLayoutPackageFinder`.

I'm not sure what's breaking because `setup.py` is not packaged in the wheel, which is the only use of `__version__.py` I can find.

Jake-Moss committed:
Remove .pxd files to where the module will end up (above cython/)

Remove paramaters.pxi, it was largely unused.  Remove .pyx source from wheels (they weren't being included anyway, not
sure why), it's not useful without our setup.py anyway. .pxd files are useful to others though.  Remove __version__.py,
not necessary
